src/Components/Facility/ConsultationForm.tsx

```diff
@@ -318,6 +318,13 @@ export const ConsultationForm = (props: any) => {
     fetchPatientName();
   }, [dispatchAction, patientId]);
 
+  useEffect(() => {
+    dispatch({
+      type: "set_form",
+      form: { ...state.form, admission_date: new Date() },
+    });
+  }, []);
+
```

Comment:
@khavinshankar but the user should be able to see the current date and time in the field of date of admission rather than the date at which the app was loaded. The user can also edit the date so if we take the `new Date()` when onSubmit then there will be no use of the custom date field which is displayed to the user.

Reply:
He meant more like: 

```
admission_date: state.form.admission_date ?? new Date()
```

And in the date time input field, you can have something like this to always show current time while the form state's admission_date being null. It would be non-null only when `onChange` is triggered.

```
value={state.form.admission_date ?? new Date()}
```

Reply:
@khavinshankar but I don't think that's a good way. It could mess up with the draft forms right?

Say the user wanted to create an admission consultation at 11 AM. But had to put the app in background for some reason, bought the app back, form drafts restored the form, but the form drafts would have saved `admission_date` as null. This would cause the admission date to be time at which the submit button was clicked although user may not have intended that.

Reply:
@rithviknishad @khavinshankar can you suggest me the final changes?

Reply:
@khavinshankar @Ashesh3 final changes ?

Reply:
I think it should be left to be done on component mount, so that drafts play nice with it. In any case, the user should correctly set and verify the admission date before submission. I do see the issue of the "time" part not being pin point accurate in this solution, but it's manually configurable and they do see it while filling out the form. (correct me here if admission_date is a hidden field)

cc: @rithviknishad @khavinshankar